app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -95,17 +95,36 @@ class AddAdditionalMetadataController @Inject() (
       } yield result
   }
 
+  private def getDependenciesOfNonSelectedOptions(dependencies: Map[String, List[FormField]], nameOfOptionSelected: List[String]) =
+    dependencies.filterNot { case (name, _) => nameOfOptionSelected.contains(name) }.flatMap { case (_, fields) => fields.map(_.fieldId) }
+
+  // scalastyle:off cyclomatic.complexity
   private def updateMetadata(updatedFormFields: List[FormField], consignmentId: UUID, fileIds: List[UUID], metadataType: String)(implicit
       request: Request[AnyContent]
   ): Future[Result] = {
 
-    val updateMetadataInputs = buildUpdateMetadataInput(updatedFormFields)
+    val updateMetadataInputs: List[UpdateFileMetadataInput] = buildUpdateMetadataInput(updatedFormFields)
 
-    val deleteMetadataNames = updatedFormFields
+    val deleteMetadataNames: Set[String] = updatedFormFields
       .collect {
         case p if !updateMetadataInputs.exists(_.filePropertyName == p.fieldId) => p.fieldId :: Nil
-        case RadioButtonGroupField(_, _, _, _, _, _, selectedOption, _, _, dependencies, _) =>
-          dependencies.removed(selectedOption).flatMap { case (_, fields) => fields.map(_.fieldId) }
+        case RadioButtonGroupField(_, _, _, _, _, _, selectedOption, _, _, _, dependencies) if dependencies.nonEmpty =>
+          val nameOfOptionSelected: List[String] = List(selectedOption)
```

Comment:
I think this can be simplified a bit. The only variable part is how you get the selection option name so is it worth adding a method to the `FormField` abstract class like `selectedOptionNames` and putting the logic in there, then the case class can be simplified quite a bit?

I've put an example here https://github.com/nationalarchives/tdr-transfer-frontend/compare/addDependenciesPropertyToFormField...dependencies-example-branch?expand=1

Feel free to ignore this, I was finding the long case statement quite difficult to read but that might just be me.

Reply:
Cool. This is a good idea! A lot cleaner, and would prevent omissions like the `DateField` one.